src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -577,7 +577,23 @@ private void RestoreUncommittedInput()
             /// </summary>
             public bool Paste()
             {
-                MoveCaretToClosestEditableBuffer();
+                if (!TextView.Selection.IsEmpty)
+                {
+                    if (CutOrDeleteSelection(isCut: false))
+                    {
+                        MoveCaretToClosestEditableBuffer();
+                    }
+                }
+                else if (IsInActivePrompt(TextView.Caret.Position.BufferPosition))
+                {
+                    MoveCaretToClosestEditableBuffer();
+                }
+
+                // selection isn't empty means none of the selected area is editable
```

Comment:
Personally, I would expect this method to return false if the caret is in a readonly region, but no one actually consumes the return value so it's not urgent.